src/components/AuthRequired.js

```diff
@@ -0,0 +1,22 @@
+import React, { Component } from 'react'
+import { Redirect, withRouter } from 'react-router'
+
+class AuthRequired extends Component {
+    constructor (props) {
+        super(props)
+    }
+
+    checkToken = () => {
```

Comment:
```
const u = new URLSearchParams(null)
u.get('asd') // null
```
Keeping in mind the above we can:
* remove the `if` and `return false` (btw: returning either null, false or `""` is a bit against the return type, it should be unified
* we can grab `lodash`'s `get` function `new URLSearchParams(get(this.props.location.search, ''))`
* we can just return `query.get('token')` - empty of null value will be `falsy` anyway

Reply:
1. Which if do you want to remove? I return false to be sure "falsy" type is unified -> I have always false, not null or "" or sth else.
2. Why should I use lodash's function when I have the same result without it?
3. You want to return empty or null because both are falsy, but in 1. you want to unified these values. So what is the proper way in your opinion? Always false or always whatever falsy? 1 and 3 look like completely different options.

Maybe you want to resolve some problems which you didn't include in your comment and I don't know what do you want to do here.